tensorflow/core/kernels/segment_reduction_ops.cc

```diff
@@ -288,27 +317,67 @@ class UnsortedSegmentSumOp : public OpKernel {
     auto output_flat = output->flat_outer_dims<T>();
 
     auto data_ptr = data.template flat<T>().data();
-    functor::UnsortedSegmentSumFunctor<Device, T, Index>()(
-        context, context->template eigen_device<Device>(), output_rows,
-        segment_ids.shape(), segment_flat, data.NumElements(), data_ptr,
-        output_flat);
+    reductionFunctor(context, context->template eigen_device<Device>(),
+                     output_rows, segment_ids.shape(), segment_flat,
+                     data.NumElements(), data_ptr, output_flat);
   }
+  functor::UnsortedSegmentBaseFunctor<Device, T, Index>& reductionFunctor;
+};
+
+template <typename Device, class T, class Index>
+class UnsortedSegmentSumOp : public UnsortedSegmentBaseOp<Device, T, Index> {
+ public:
+  explicit UnsortedSegmentSumOp(OpKernelConstruction* context)
+      : UnsortedSegmentBaseOp<Device, T, Index>(
+            context,
+            sumFunctor) {}
+    functor::UnsortedSegmentSumFunctor<Device, T, Index> sumFunctor;
+};
+
+template <typename Device, class T, class Index>
+class UnsortedSegmentMaxOp : public UnsortedSegmentBaseOp<Device, T, Index> {
+ public:
+  explicit UnsortedSegmentMaxOp(OpKernelConstruction* context)
+      : UnsortedSegmentBaseOp<Device, T, Index>(
+            context,
+            maxFunctor) {}
+    functor::UnsortedSegmentMaxFunctor<Device, T, Index> maxFunctor;
```

Comment:
Can you make these private members and following the naming convention: 'max_functor_'.  Same for 'sumFunctor' and 'reductionFunctor' above?